### PR TITLE
Implemented cookie saving on server-side

### PR DESCRIPTION
**Pull request by pwnsdx**
+ Fixed compatibility with store-engine

### CODE REVIEW

File: src/main/util/cookie.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        };
+  });
+};
+
+const setInternalCookie = (zuid: string, expiration: Date, engine: CRUDEngine): Promise<string> => {
+  return engine.create(TABLE_NAME, COOKIE_PRIMARY_KEY, {zuid, expiration});
+};
+
+export const getInternalCookie = (response: AxiosResponse, engine: CRUDEngine): Promise<AccessTokenData> => {
+  let payload: AxiosResponse;
```
Comment (gregor) [inline src/main/util/cookie.ts]: No need for an extra let if you only set it to a parameter

File: src/main/util/cookie.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+  return Promise.resolve(response)
+    .then((response: AxiosResponse) => {
+      payload = response;
+      return getInternalCookieHelper(engine);
+    })
+    .then(({isExpired}: {isExpired: boolean}) => {
+      if (isExpired && (payload.headers && payload.headers['set-cookie'])) {
+        const cookies = payload.headers['set-cookie'].map(Cookie.parse);
+        for (let cookie of cookies) {
```
Comment (gregor) [inline src/main/util/cookie.ts]: Should be const - you are never changing it.

File: index.js [hunk 3]
```diff
@@ -4,27 +4,41 @@ const argv = require('optimist')
...
   console.log('Received notification via WebSocket', notification);
 });
 
-client.login(login)
-  .then((context) => {
+Promise.resolve()
+  .then(() => {
+    // Trying to login (works only if there is already a valid cookie stored in the FileEngine)
+    return client.init(login.email);
+  }).catch((error) => {
```
Comment (gregor) [inline index.js]: Different code style? Then and catch are usually on new lines.

File: src/main/util/cookie.ts [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+  });
+};
+
+const setInternalCookie = (zuid: string, expiration: Date, engine: CRUDEngine): Promise<string> => {
+  return engine.create(TABLE_NAME, COOKIE_PRIMARY_KEY, {zuid, expiration});
+};
+
+export const retrieveCookie = (response: AxiosResponse, engine: CRUDEngine): Promise<AccessTokenData> => {
+  return loadExistingCookie(engine).then(({isExpired}: {isExpired: boolean}) => {
+    if (isExpired && (response.headers && response.headers['set-cookie'])) {
```
Comment (gregor) [inline src/main/util/cookie.ts]: Why do we only do this with expired cookies?

**Comment by bennyn** [comment]
Thanks for the review. Issues are fixed now. 😎

**Pull request merged**